tests/test_cases/test_array_simple/test_array_simple.py

```diff
@@ -0,0 +1,197 @@
+
+tlog = logging.getLogger("cocotb.test")
+
+def _check_value(tlog, hdl, expected):
+    if hdl.value != expected:
+        raise TestFailure("{2!r}: Expected >{0}< but got >{1}<".format(expected, hdl.value, hdl))
+    else:
+        tlog.info("   Found {0!r} ({1}) with value={2}".format(hdl, hdl._type, hdl.value))
+
+
+@cocotb.test()
+async def test_1dim_array_handles(dut):
+    """Test getting and setting array values using the handle of the full array."""
+
+    cocotb.fork(Clock(dut.clk, 1000, 'ns').start())
+
+    # Set values with '<=' operator
+    dut.array_7_downto_4 <= [0xF0, 0xE0, 0xD0, 0xC0]
+    dut.array_4_to_7     <= [0xB0, 0xA0, 0x90, 0x80]
+    dut.array_3_downto_0 <= [0x70, 0x60, 0x50, 0x40]
+    dut.array_0_to_3     <= [0x30, 0x20, 0x10, 0x00]
+
+    await Timer(1000, 'ns')
+
+    _check_value(tlog, dut.array_7_downto_4, [0xF0, 0xE0, 0xD0, 0xC0])
+    _check_value(tlog, dut.array_4_to_7    , [0xB0, 0xA0, 0x90, 0x80])
+    _check_value(tlog, dut.array_3_downto_0, [0x70, 0x60, 0x50, 0x40])
+    _check_value(tlog, dut.array_0_to_3    , [0x30, 0x20, 0x10, 0x00])
+
+    # Set values through HierarchyObject.__setattr__ method on 'dut' handle
+    dut.array_7_downto_4 = [0x00, 0x11, 0x22, 0x33]
+    dut.array_4_to_7     = [0x44, 0x55, 0x66, 0x77]
+    dut.array_3_downto_0 = [0x88, 0x99, 0xAA, 0xBB]
+    dut.array_0_to_3     = [0xCC, 0xDD, 0xEE, 0xFF]
+
+    await Timer(1000, 'ns')
+
+    _check_value(tlog, dut.array_7_downto_4, [0x00, 0x11, 0x22, 0x33])
+    _check_value(tlog, dut.array_4_to_7    , [0x44, 0x55, 0x66, 0x77])
+    _check_value(tlog, dut.array_3_downto_0, [0x88, 0x99, 0xAA, 0xBB])
+    _check_value(tlog, dut.array_0_to_3    , [0xCC, 0xDD, 0xEE, 0xFF])
+
+
+@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith(("icarus", "ghdl")))
+async def test_ndim_array_handles(dut):
+    """Test getting and setting multi-dimensional array values using the handle of the full array."""
+
+    cocotb.fork(Clock(dut.clk, 1000, 'ns').start())
+
+    # Set values with '<=' operator
+    dut.array_2d <= [
+        [0xF0, 0xE0, 0xD0, 0xC0],
+        [0xB0, 0xA0, 0x90, 0x80]
+    ]
+
+    await Timer(1000, 'ns')
+
+    _check_value(tlog, dut.array_2d, [[0xF0, 0xE0, 0xD0, 0xC0], [0xB0, 0xA0, 0x90, 0x80]])
+
+    # Set values through HierarchyObject.__setattr__ method on 'dut' handle
+    dut.array_2d = [
+        [0x00, 0x11, 0x22, 0x33],
+        [0x44, 0x55, 0x66, 0x77]
+    ]
+
+    await Timer(1000, 'ns')
+
+    _check_value(tlog, dut.array_2d, [[0x00, 0x11, 0x22, 0x33], [0x44, 0x55, 0x66, 0x77]])
+
+
+@cocotb.test()
+async def test_1dim_array_indexes(dut):
+    """Test getting and setting values of array indexes."""
+
+    cocotb.fork(Clock(dut.clk, 1000, 'ns').start())
+
+    dut.array_7_downto_4 <= [0xF0, 0xE0, 0xD0, 0xC0]
+    dut.array_4_to_7     <= [0xB0, 0xA0, 0x90, 0x80]
+    dut.array_3_downto_0 <= [0x70, 0x60, 0x50, 0x40]
+    dut.array_0_to_3     <= [0x30, 0x20, 0x10, 0x00]
+
+    await Timer(1000, 'ns')
+
+    # Check indices
```

Comment:
```suggestion
    # Check indexes
```
?

Reply:
I've always seen **indices** in a mathematics context relating to arrays or matrices.

Reply:
Makes sense.